java/lang/String/String.cpp

```diff
@@ -845,4 +850,153 @@ boolean String::operator>=(const String &target) const {
+    if (length > 0)
+        result = String(buffer, length);
+    return result;
+}
+
+String String::print(const String &format, unsigned int value) {
+    String result;
+    char buffer[256] = {0};
+    const int length = snprintf(buffer, sizeof(buffer), format.toString(), value);
+    if (length > 0)
+        result = String(buffer, length);
+    return result;
+}
+
+String String::print(const String &format, unsigned long value) {
+    String result;
+    char buffer[256] = {0};
+    const int length = snprintf(buffer, sizeof(buffer), format.toString(), value);
+    if (length > 0)
+        result = String(buffer, length);
+    return result;
+}
+
+String String::print(const String &format, double value) {
+    String result;
+    char buffer[256] = {0};
+    const int length = snprintf(buffer, sizeof(buffer), format.toString(), value);
+    if (length > 0)
+        result = String(buffer, length);
+    return result;
+}
+
+String String::print(const String &format, float value) {
+    String result;
+    char buffer[256] = {0};
+    const int length = snprintf(buffer, sizeof(buffer), format.toString(), value);
+    if (length > 0)
+        result = String(buffer, length);
+    return result;
+}
+
+String String::print(const String &format, char *value) {
+    String result;
+    char buffer[256] = {0};
+    const int length = snprintf(buffer, sizeof(buffer), format.toString(), value);
+    if (length > 0)
+        result = String(buffer, length);
+    return result;
+}
+
+String String::print(const String &format, Short value) {
+    return String::print(format, value.shortValue());
+}
+
+String String::print(const String &format, Integer value) {
+    return String::print(format, value.intValue());
+}
+
+String String::print(const String &format, Long value) {
+    return String::print(format, value.longValue());
+}
+
+String String::print(const String &format, Float value) {
+    return String::print(format, value.floatValue());
+}
+
+String String::print(const String &format, Double value) {
+    return String::print(format, value.doubleValue());
+}
+
+String String::print(const String &format, String value) {
+    return String::print(format, value.toString());
+}
+
+String String::format(const String &format) {
+    const String pattern = "%([[:digit:]]+)?([-#+0]*)?([[:digit:]]+)?(\\.[[:digit:]]+)?([diuoxXfFeEgGaAcspn%])";
+    String result;
+    String inputString(format);
+    string inputStringPtr = inputString.toString();
+    int inputStringLength = inputString.getSize();
+    int inputStringOffset = 0;
+    int errorCode = 0;
+    regex_t regex;
+
+    errorCode = regcomp(&regex, pattern.toString(), REG_EXTENDED);
+    while (errorCode == 0 && inputStringOffset < inputString.getSize()) {
+        regmatch_t matchedResult[16] = {0}; // max 16 groups
+        errorCode = regexec(&regex, inputStringPtr, 16, matchedResult, 0);
+        if (errorCode == 0) {

```

Comment:
Please help me shorten this with return early
`if (errorCode != 0) { ...; break;}`